testsuite/classlibrary/TestPmono.sc

```diff
@@ -0,0 +1,45 @@
+
+TestPmono : UnitTest {
+	var server;
+
+	setUp {
+		server = Server(this.class.name);
+	}
+
+	tearDown {
+		server.quit;
+		server.remove;
+	}
+
+	test_Pmono_makes_node_after_pause {
+		var player, event;
+		this.bootServer;
+		player = Pmono(
+			\default,
+			\degree, Pseries(-7, 1, inf),
+			\dur, 0.01
+		).play;
+		// we are simulating the user action of
+		// stopping an EventStreamPlayer while it's running
+		// there is really no way to do that except a hardcoded wait
+		0.05.wait;
```

Comment:
tests still pass without waiting.. if the only thing being checked here is the state of the Pmono after its player is stopped (which is what i infer from the assert message) then the rest of this seems unnecessary. am i missing something?

Reply:
It's not a complete test unless it goes through all of these stages:

1. An event is generated and played. This does a couple of things:
   - Set up the EventStreamCleanup (which will clear the node when the player is stopped).
   - Playing the event grabs a node ID and communicates it back to the PmonoStream.
2. The player is stopped. This triggers the EventStreamCleanup, clearing the `id` variable, so that...
3. The next event sees that there's no node ID and produces a `\monoNote` event, just like step 1.

If the wait is removed altogether, then `stop` will be called before the test thread yields -- so steps 1 and 2 will not occur until *after* checking step 3. That's not a valid test. The test is to initialize normally, *then* stop (de-initialize), then see what will happen after that. If you don't init and de-init, then you're testing normal init, not re-init. Normal init was always working, so of course it passes (and in that case, it would pass without the fix = false pass = invalid test).

If there's a `0.wait`, it *might* go through the whole sequence but I'm not totally sure of that.

So I think the best thing is to reduce the wait time to, oh, 1 ms, to be absolutely sure `stop` happens after an event plays.